### DISCUSSION TITLE
v5: Middleware not working with Route.resource

### DISCUSSION THREAD

**Discussion by mayureshwaykole** [adonisjs/core#1469] [Help] [answered]
**version**:
Adonisjs v5

**Issue**
I am using Route.resource and adding a middleware as mentioned in the docs
`Route.resource('/user', 'UsersController').apiOnly().middleware('auth')`
This doesnt seem to be working on v5.

Based on the type prompts looks like It's expecting something like this
`Route.resource('/user', 'UsersController').apiOnly().middleware({index: 'auth', create:'auth'})`

Is this a bug or am I missing something ?

**Comment by thetutlage** [comment, marked as answer]
A resource generates a total of 7 routes, you have to tell on which route to apply the middleware.

If it's all the routes. Then simply use the `*` keyword.

```ts
Route.resource().middleware({ '*': 'auth' })
```

**Reply by farshadfahimi** [reply]
@thetutlage 
Is it possible on v4?
